templates/registration/registration_form.html

```diff
@@ -39,7 +39,7 @@ <h2 class="align-right">{% trans "Create a new account" %}</h2>
                    id="Username"
                    placeholder="{% trans "Username" %}"
                    name="username"
-                   value="{{ form.cleaned_data.username }}">
+                   value="{{ form.cleaned_data.username|default:form.username.value }}">
```

Comment:
```suggestion
                   value="{{ form.cleaned_data.username|default:form.username.value|default:"" }}">
```
Avoid "None" in form field, hiding "placeholder" value and confusing users.